MPChartLib/src/main/java/com/github/mikephil/charting/renderer/LineChartRenderer.java

```diff
@@ -122,6 +122,14 @@ protected void drawDataSet(Canvas c, ILineDataSet dataSet) {
 
         mRenderPaint.setStrokeWidth(dataSet.getLineWidth());
         mRenderPaint.setPathEffect(dataSet.getDashPathEffect());
+        mRenderPaint.setStrokeCap(dataSet.getStrokeCap());
+    
```

Comment:
Why did you have to move this part out of the `drawHorizontalBezier` function? What effect did the shader/colour have on the calculations of the path?

Reply:
You mean `setStrokeCap` ? That was basically the reason for a bug. Before the strokeCap was set to `Round` inside the `drawHorizontalBezier`. That led to the behaviour that the line was dotted when one of the lines was cubier. This was very weird behaviour and I didn't understand why that happened at first.

Now the `strokeCap` can be set from the outside and it can apply to each lineStyle to ensure no surprising behaviour. The same goes actually also for the shader. I think when it can be set, it should apply to all styles, not only the one we use. I hope this answers the question :)